### PR TITLE
refactor: Move lockfile to a separate crate

### PR DESCRIPTION
**Pull request by bartlomieju**
Moves the lockfile implementation to a separate crate so other projects
like Deploy can use it as well.

### CODE REVIEW

**Review by ry — APPROVED** [review]
LGTM

File: cli/args/lockfile.rs [hunk 1]
```diff
@@ -1,602 +1,84 @@
...
+        id: id.as_serialized(),
+      })
+      .collect();
+
+    NpmPackageLockfileInfo {
+      display_id: self.id.display(),
+      serialized_id: self.id.as_serialized(),
+      integrity: self.dist.integrity().to_string(),
+      dependencies,
+    }
```
Comment (dsherret) [inline cli/args/lockfile.rs:82]: Slightly unfortuante that the common case requires cloning now, but not a big deal. We can fix this up when extracting stuff like `NpmPackageId` from the CLI repo.

**Review by dsherret — APPROVED** [review]
LGTM

File: cli/args/lockfile.rs (hunk 1 above)
Comment (bartlomieju) [reply cli/args/lockfile.rs:82]: I do agree, ideally we wouldn't require any clones for the lifetime of the lockfile checking

**Pull request merged**